Dependency Extraction Fails for Gradle < 6

As reported in https://github.com/renovatebot/renovate/issues/4471:

The new gradle extraction script depends on a type that was introduced with Gradle 6, leading to error like this when it is executed on against a project with a Gradl 5 wrapper:

```
FAILURE: Build failed with an exception.\n\n* Where:\nInitialization script '/mnt/renovate/gh/moltenbits/renovate-gradle-bug/renovate-plugin.gradle' line: 4\n\n* What went wrong:\nCould not compile initialization script '/mnt/renovate/gh/moltenbits/renovate-gradle-bug/renovate-plugin.gradle'.\n> startup failed:\n  initialization script '/mnt/renovate/gh/moltenbits/renovate-gradle-bug/renovate-plugin.gradle': 4: unable to resolve class org.gradle.api.artifacts.repositories.UrlArtifactRepository\n   @ line 4, column 1.\n     import org.gradle.api.artifacts.repositories.UrlArtifactRepository
```

--- a/lib/manager/gradle/gradle-updates-report.ts
+++ b/lib/manager/gradle/gradle-updates-report.ts
@@ -32,7 +32,7 @@ async function createRenovateGradlePlugin(localDir: string): Promise<void> {
   const content = `
 import groovy.json.JsonOutput
 import org.gradle.api.internal.artifacts.dependencies.DefaultExternalModuleDependency
-import org.gradle.api.artifacts.repositories.UrlArtifactRepository
+import java.net.URI
 import java.util.concurrent.ConcurrentLinkedQueue
 
 def output = new ConcurrentLinkedQueue<>();
@@ -43,7 +43,7 @@ allprojects {
         def project = ['project': project.name]
         output << project
         def repos = (repositories + buildscript.repositories + settings.pluginManagement.repositories)
-           .findAll { it instanceof UrlArtifactRepository && it.url.scheme ==~ /https?/ }
+           .findAll { it instanceof URI && it.url.scheme ==~ /https?/ }
            .collect { "$it.url" }
            .unique()
         project.repositories = repos

